src/validators/string.rs

```diff
@@ -93,17 +93,23 @@ impl Validator for StrConstrainedValidator {
         if self.strip_whitespace {
             str = str.trim();
         }
+
+        let mut str_len: Option<usize> = None;
+        if (self.min_length, self.max_length) != (None, None) {
+            str_len = Some(str.chars().count());
+        }
```

Comment:
```suggestion
        let str_len: Option<usize> = if self.min_length.is_some() | self.max_length.is_some() {
            Some(str.chars().count())
        } else {
            None
        };
```

is optionally easier to read.